src/main/java/org/candlepin/insights/orgsync/OrgSyncConfiguration.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.insights.orgsync;
+
+import org.quartz.JobDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.scheduling.quartz.CronTriggerFactoryBean;
+import org.springframework.scheduling.quartz.JobDetailFactoryBean;
+
+/**
+ * Configuration class for beans related to the OrgSyncJob.  Includes several conditional beans which are
+ * loaded based on a value set in the application properties.  Picked up via the @ComponentScan that's
+ * implicit in @SpringBootApplication on the {@link org.candlepin.insights.BootApplication} class.
+ */
+@EnableConfigurationProperties(OrgSyncProperties.class)
+@Configuration
+@PropertySource("classpath:/rhsm-conduit.properties")
+public class OrgSyncConfiguration {
+    public static final String ORG_SYNC_STRATEGY = "rhsm-conduit.org-sync.strategy";
+    private static final Logger log = LoggerFactory.getLogger(OrgSyncProperties.class);
+
+    @Autowired
+    private OrgSyncProperties orgSyncProperties;
+
+    @Bean
+    @ConditionalOnProperty(value = ORG_SYNC_STRATEGY, havingValue = "fileBasedOrgListStrategy")
+    @ConfigurationProperties(prefix = "rhsm-conduit.org-sync.file-based-org-list-strategy")
+    public FileBasedOrgListStrategyProperties fileBasedOrgListStrategyProperties() {
+        return new FileBasedOrgListStrategyProperties();
+    }
+
+    @Bean
+    @ConditionalOnProperty(value = ORG_SYNC_STRATEGY, havingValue = "fileBasedOrgListStrategy")
+    public OrgListStrategy fileBasedOrgListStrategy(FileBasedOrgListStrategyProperties conf) {
+        log.info("Using {} strategy", orgSyncProperties.getStrategy());
+        return new FileBasedOrgListStrategy(conf);
+    }
+
+    @Bean
+    @Conditional(NoStrategyCondition.class)
+    public OrgListStrategy noOrgListStrategy() {
```

Comment:
Did you experiment with omitting this implementation altogether? I would think the application can simply throw an exception by virtue of trying to inject an `OrgListStrategy` and if the other condition isn't satisfied, it'll fail to resolve the `OrgListStrategy`...

If we still want friendly message there, it looks like another option is to use `@Autowired(required=false)`, and then in the constructor of OrgSyncJob, throw this exception if the strategy is null.

Either of these options would allow us to drop `NoStrategyCondition`.

Reply:
The problem is that the injection doesn't occur at application initialization; it occurs when an instance of the job is instantiated.  With `NoStrategyCondition` deployment fails immediately with the message

```
Caused by: java.lang.IllegalStateException: Either no value or no valid value is defined for rhsm-conduit.org-sync.strategy
```

Without it, deployment *succeeds* but the application collapses a few minutes later (or however long the job period is) when the job runs with the following opaque message

```
No qualifying bean of type 'org.candlepin.insights.orgsync.OrgListStrategy' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
```

Using the `Autowired(required=false)` approach followed by throwing an exception on null would lead to a similar problem: the application would deploy successfully only to blow up later.  It'd rather give immediate feedback that the application is misconfigured.

Reply:
I just opened #47 with an alternate approach that I think is a touch cleaner (but functionally the same). Please take a look and let me know what you think.

Reply:
Merged!